src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1872,4 +1879,128 @@ sym2.ind (λ v w, is_bridge_iff_adj_and_forall_cycle_not_mem) e
 
 end bridge_edges
 
+section induced
+
+/-! ### Connectivity of induced graphs -/
+
+lemma induce_singleton_connected (v : V) :
+  (G.induce {v}).connected :=
+begin
+  rw connected_iff,
+  refine ⟨_, by simp⟩,
+  rintro ⟨x, hx⟩ ⟨y, hy⟩,
+  rw set.mem_singleton_iff at hx hy,
+  subst_vars,
+end
+
+lemma induce_union_connected {s t : set V}
+  (sconn : (G.induce s).connected) (tconn : (G.induce t).connected) (sintert : (s ∩ t).nonempty ) :
+  (G.induce $ s ∪ t).connected :=
+begin
+  obtain ⟨u, hus, hut⟩ := sintert,
+  rw connected_iff_basepoint,
+  use ⟨u, or.inl hus⟩,
+  rintro ⟨v, hv|hv⟩,
+  { obtain ⟨p⟩ := (sconn ⟨u, hus⟩ ⟨v, hv⟩),
+    exact ⟨p.map (induce_hom_of_le le_sup_left : (G.induce s) →g (G.induce $ s ⊔ t))⟩, },
+  { obtain ⟨p⟩ := (tconn ⟨u, hut⟩ ⟨v, hv⟩),
+    exact ⟨p.map (induce_hom_of_le le_sup_right : (G.induce t) →g (G.induce $ s ⊔ t))⟩, },
+end
+
+lemma induce_pair_connected_of_adj {u v : V} (huv : G.adj u v) :
+  (G.induce {u, v}).connected :=
+begin
+  rw connected_iff,
+  refine ⟨_, by simp⟩,
+  rintro ⟨x, hx⟩ ⟨y, hy⟩,
+  simp only [set.mem_insert_iff, set.mem_singleton_iff] at hx hy,
+  obtain rfl|rfl := hx; obtain rfl|rfl := hy;
+    refl <|> { refine ⟨walk.cons _ walk.nil⟩, simp [huv, huv.symm] }
+end
+
+lemma induce_connected_adj_union {s t : set V}
+  (sconn : (G.induce s).connected) (tconn : (G.induce t).connected) {v w} (hv : v ∈ s) (hw : w ∈ t)
+  (a : G.adj v w) : (G.induce $ s ∪ t).connected :=
+begin
+  have : s ∪ t = s ∪ {v, w} ∪ t, by
+  { rw [set.union_comm s {v, w}, set.union_assoc], symmetry,
+    apply set.union_eq_self_of_subset_left,
+    simp only [set.insert_subset, set.singleton_subset_iff, hv, hw, set.mem_union, true_or,
+               or_true, and_self], },
+  rw this,
+  refine induce_union_connected
+    (induce_union_connected sconn (induce_pair_connected_of_adj a) _) tconn _,
+  { refine ⟨v, hv, _⟩, simp, },
+  { refine ⟨w, or.inr _, hw⟩, simp, }
+end
```

Comment:
It would be nice if this were more direct from lemmas about subgraphs. Here's some partial progress:
```lean
lemma subgraph_of_adj_le_iff {u v : V} (h : G.adj u v) (H : G.subgraph) :
  G.subgraph_of_adj h ≤ H ↔ u ∈ H.verts ∧ v ∈ H.verts ∧ H.adj u v :=
begin
  sorry,
end

lemma subgraph_of_adj_le_induce_iff {u v : V} (h : G.adj u v) (H : G.subgraph) (s : set V) :
  G.subgraph_of_adj h ≤ H.induce s ↔ u ∈ s ∧ v ∈ s ∧ H.adj u v :=
by simp [subgraph_of_adj_le_iff] { contextual := tt }

lemma subgraph.induce_sup_subgraph_of_adj_eq {u v : V} {H : G.subgraph}
  (h : H.adj u v) {s : set V} (hu : u ∈ s) (hv : v ∈ s) :
  H.induce s ⊔ G.subgraph_of_adj (H.adj_sub h) = H.induce s :=
begin
  rw [eq_comm, left_eq_sup],
  simp [subgraph_of_adj_le_induce_iff, *],
end

lemma subgraph.induce_connected_adj_union {H : G.subgraph} {s t : set V}
  (sconn : (H.induce s).connected) (tconn : (H.induce t).connected) {v w} (hv : v ∈ s) (hw : w ∈ t)
  (a : H.adj v w) : (H.induce $ s ∪ t).connected :=
begin
  apply subgraph.connected_of_connected_le
    (H.induce s ⊔ H.induce t ⊔ G.subgraph_of_adj (H.adj_sub a)),
  { transitivity H.induce (s ∪ t) ⊔ G.subgraph_of_adj (H.adj_sub a),
    { exact sup_le_sup_right (sup_induce_le_induce_union H s t) _, },
    rw subgraph.induce_sup_subgraph_of_adj_eq a; simp [*] },
  { simp [set.insert_eq_of_mem, hv, hw], },
  -- ⊢ (H.induce s ⊔ H.induce t ⊔ G.subgraph_of_adj _).connected
  sorry
end

lemma induce_connected_adj_union {s t : set V}
  (sconn : (G.induce s).connected) (tconn : (G.induce t).connected) {v w} (hv : v ∈ s) (hw : w ∈ t)
  (a : G.adj v w) : (G.induce $ s ∪ t).connected :=
begin
  rw simple_graph.induce_eq_coe_induce_top at sconn tconn ⊢,
  exact subgraph.induce_connected_adj_union sconn tconn hv hw a,
end
```

Reply:
I skipped `subgraph.induce_connected_adj_union` since I'm not really convince by it's use.